src/Microsoft.ML.Data/Transforms/ColumnSelecting.cs

```diff
@@ -394,13 +394,15 @@ public static IDataView Create(IHostEnvironment env, ModelLoadContext ctx, IData
             return transform.Transform(input);
         }
 
-        public static IDataTransform CreateKeep(IHostEnvironment env, IDataView input, string[] keepColumns, bool keepHidden = false)
+        [BestFriend]
+        internal static IDataTransform CreateKeep(IHostEnvironment env, IDataView input, string[] keepColumns, bool keepHidden = false)
         {
             var transform = new ColumnSelectingTransformer(env, keepColumns, null, keepHidden);
             return new SelectColumnsDataTransform(env, transform, new Mapper(transform, input.Schema), input);
         }
 
-        public static IDataTransform CreateDrop(IHostEnvironment env, IDataView input, params string[] dropColumns)
+        [BestFriend]
+        internal static IDataTransform CreateDrop(IHostEnvironment env, IDataView input, params string[] dropColumns)
```

Comment:
In my PR, I am making these private :)

Reply:
I think @rogancarr wanted mlContext.SchemaOperations catalog, so I would definitely add them to that catalog. 
But I don't understand how you can make them private.

btw, I should change type from `IDataTransform` to `IDataView`.

---
In reply to: [253240652](https://github.com/dotnet/machinelearning/pull/2383#discussion_r253240652) [](ancestors = 253240652)

Reply:
(I replaced them with new Transformer().Transform())

---
In reply to: [253245274](https://github.com/dotnet/machinelearning/pull/2383#discussion_r253245274) [](ancestors = 253245274,253240652)